openmdao/utils/assert_utils.py

```diff
@@ -35,16 +36,19 @@ def assert_warning(category, msg):
     AssertionError
         If the expected warning is not raised.
     """
+    # print(MPI.COMM_WORLD.rank)
+    rank = MPI.COMM_WORLD.rank if MPI is not None else 0
     with reset_warning_registry():
         with warnings.catch_warnings(record=True) as w:
             warnings.simplefilter("always")
             yield
 
-    for warn in w:
-        if (issubclass(warn.category, category) and str(warn.message) == msg):
-            break
-    else:
-        raise AssertionError("Did not see expected %s: %s" % (category.__name__, msg))
+    if rank == 0:
```

Comment:
I think it would be better to keep assert_warning blissfully unaware of MPI, and handle any rank related behaviors in your unit test.  There could be instances where a warning might be generated on some other rank than rank 0 and a test might want to check that.